### PR TITLE
47728 VA Request Flow - Best Times to Contact (Acheron)

### LINKED ISSUE #47728: VA Request Flow - Best Times to Contact (Acheron)
## Description
Scheduling Manager is the service layer used to send the data for the VA request. There is a plan to sunset scheduling manager later this year. We need to remove the best times to contact selection on the frontend since the data is no longer supported in the Acheron service. 

---

## Acceptance Criteria
Background: the `va_online_scheduling_acheron_service` toggle is on

Given the user is in the VA request flow 
When the user is on the Contact Information page
Then the user will no longer see the option to select best times to contact
And page matches [copy doc](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/appointments/va-online-scheduling/content/copy-docs/contact-information.md#request-flow-specific)

Given the user is in the VA request flow 
When the user is on the Review Details page
Then the user will no longer see the option to select best times to contact 
And page matches [copy doc](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/appointments/va-online-scheduling/content/copy-docs/review-details.md#va-request)

Given the user is in the VA request flow 
When the user is on the Pending Confirmation page
Then the user will no longer see the option to select best times to contact 
And page matches [copy doc](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/appointments/va-online-scheduling/content/copy-docs/appointment-details.md#va-request-confirmation-state)

Given the user is in the Pending list
When the user clicks on details for their pending appointment 
Then the user will no longer see the best times to contact 
And page matches [copy doc for pending appointment](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/appointments/va-online-scheduling/content/copy-docs/appointment-details.md#va-request-pending)
And page matches [copy doc for canceled pending appointment](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/appointments/va-online-scheduling/content/copy-docs/appointment-details.md#va-request-canceled)

## Definition of Done
- [ ] All acceptance criteria are met
- [ ] Documentation is updated

### PR DESCRIPTION
**Pull request by vbahinwillit**
## Description
This PR hides best time to call information when acheron feature flag is on

## Original issue(s)
fixes department-of-veterans-affairs/va.gov-team#47728

## Testing done
Unit testing

## Acceptance criteria
- [ ] All test past

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by jenniemc — CHANGES_REQUESTED** [review]
Are we suppose to remove the Best Time to Call question from the VA Request flow and Community Care request flow? The AC is for VA Request but the enhancement is also removing the question from the Community Care flow. 

Looks like the review page shows best time to call info even though none is entered.

**Comment by vbahinwillit** [comment]
> 

Verified with Peter that changes should be for VA appointment requests only.

File: src/applications/vaos/new-appointment/components/ReviewPage/ReviewDirectScheduleInfo.jsx [hunk 1]
```diff
@@ -30,7 +31,14 @@ export default function ReviewDirectScheduleInfo({
       {facility.address?.city}, <State state={facility.address?.state} />
       <ReasonForAppointmentSection data={data} />
       <hr aria-hidden="true" className="vads-u-margin-y--2" />
-      <ContactDetailSection data={data} flowType={FLOW_TYPES.DIRECT} />
```
Comment (vbahinwillit) [inline src/applications/vaos/new-appointment/components/ReviewPage/ReviewDirectScheduleInfo.jsx:33]: Removed 'flowType' prop to match other calls to the component that don't pass 'flowType'. The information can be retrieved via useSelector hook.

File: src/applications/vaos/new-appointment/components/ContactInfoPage.jsx [hunk 2]
```diff
@@ -136,11 +140,17 @@ export default function ContactInfoPage() {
...
+          : [],
       'ui:options': {
         showFieldLabel: true,
         classNames: 'vaos-form__checkboxgroup',
-        hideIf: () => flowType === FLOW_TYPES.DIRECT,
+        hideIf: () => {
+          if (featureAcheronService)
+            return userData.facilityType === FACILITY_TYPES.VAMC;
+          return flowType === FLOW_TYPES.DIRECT;
+        },
```
Comment (vbahinwillit) [inline src/applications/vaos/new-appointment/components/ContactInfoPage.jsx:153]: Hide best time to call checkboxes for both direct and request flows when acheron flag is true.

**Review by jenniemc — APPROVED** [review]
LGTM! And I was able to see the changes in the review instance!

**Pull request merged**